app/models/concerns/assignable.rb

```diff
@@ -114,23 +113,31 @@ def after_user_group_assignment_changed(user_group_assignment = nil)
       # Will be called when an assignment is changed (save/destroy) for the assignable model.
     end
 
-    def create_users_assignments
-      return if skip_user_assignments
-
-      role = if top_level_assignable?
-               UserRole.find_predefined_owner_role
-             else
-               permission_parent.role_for_user(created_by, team)
-             end
-
-      UserAssignment.create!(
-        user: created_by,
-        assignable: self,
-        assigned: top_level_assignable? ? :manually : :automatically,
-        user_role: role
-      )
+    def create_user_assignments!(user = created_by)
+      # First create initial assignments for the object's creator
+      if top_level_assignable?
```

Comment:
maybe convert this to guard clauses to reduce nesting of if statements, so something like:

```
if top_level_assignable?
  user_assignments.create!(user: user, assigned: :manually, user_role: UserRole.find_predefined_owner_role)
  return
end

parent_assignment = permission_parent.user_assignments.find_by(user: user, team: team)
if parent_assignment.present?
  user_assignments.create!(user: user, user_role: parent_assignment.user_role)
  return
end

...
```